Build failure: ripes

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

- `nix build nixpkgs#ripes`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/282686955

### Relevant log output

```console
In file included from /tmp/nix-build-ripes-2.2.6-unstable-2024-04-04.drv-0/source/external/VSRTL/graphics/vsrtl_componentgraphic.cpp:14:
In file included from /tmp/nix-build-ripes-2.2.6-unstable-2024-04-04.drv-0/source/external/VSRTL/external/cereal/include/cereal/archives/json.hpp:56:
/tmp/nix-build-ripes-2.2.6-unstable-2024-04-04.drv-0/source/external/VSRTL/external/cereal/include/cereal/external/rapidjson/document.h:319:82: error: cannot assign to non-static data member 'length' with const-qualified type 'const SizeType' (aka 'const unsigned int')
  319 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
      |                                                                           ~~~~~~ ^
/tmp/nix-build-ripes-2.2.6-unstable-2024-04-04.drv-0/source/external/VSRTL/external/cereal/include/cereal/external/rapidjson/document.h:325:20: note: non-static data member 'length' declared const here
  325 |     const SizeType length; //!< length of the string (excluding the trailing NULL terminator)
      |     ~~~~~~~~~~~~~~~^~~~~~
[ 55%] Building CXX object external/VSRTL/graphics/CMakeFiles/vsrtl_gfx_lib.dir/vsrtl_gridcomponent.cpp.o
1 error generated.
make[2]: *** [external/VSRTL/graphics/CMakeFiles/vsrtl_gfx_lib.dir/build.make:139: external/VSRTL/graphics/CMakeFiles/vsrtl_gfx_lib.dir/vsrtl_componentgraphic.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:990: external/VSRTL/graphics/CMakeFiles/vsrtl_gfx_lib.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```


### Additional context

_No response_

### System metadata

 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.3.0, macOS 15.3.1`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Lix, like Nix) 2.92.0
System type: aarch64-darwin
Additional system types: x86_64-darwin
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /Users/WeetHet/.config/nix/nix.conf:/Users/WeetHet/.nix-profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/WeetHet/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/q32av18qnal3nf3s8ly4mnphy655s17j-lix-2.92.0/share`
 - nixpkgs: `/nix/store/6mkc4yfr19a6b0m3gmig1v3yr6jbi3cj-source

### Notify maintainers

@rewine
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [X] I assert that this is a bug and not a support request.
- [X] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [X] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ri/ripes/package.nix
+++ b/pkgs/by-name/ri/ripes/package.nix
@@ -4,6 +4,7 @@
   fetchFromGitHub,
   pkg-config,
   qt6,
+  cereal,
   cmake,
   python3,
   unstableGitUpdater,
@@ -22,6 +23,12 @@ stdenv.mkDerivation rec {
     hash = "sha256-aNJTM/s4GNhWVXQxK1R/rIN/NmeKglibQZMh8ENjIzo=";
   };
 
+  postPatch = ''
+    rm -r external/VSRTL/external/cereal
+    substituteInPlace {src/serializers.h,src/io/iobase.h} \
+      --replace-fail "VSRTL/external/cereal/include/cereal/cereal.hpp" "cereal/cereal.hpp"
+  '';
+
   nativeBuildInputs = [
     cmake
     pkg-config
@@ -30,6 +37,7 @@ stdenv.mkDerivation rec {
   ];
 
   buildInputs = [
+    cereal
     qt6.qtbase
     qt6.qtsvg
     qt6.qtcharts

--- a/pkgs/by-name/ri/ripes/package.nix
+++ b/pkgs/by-name/ri/ripes/package.nix
@@ -3,10 +3,7 @@
   stdenv,
   fetchFromGitHub,
   pkg-config,
-  qtbase,
-  qtsvg,
-  qtcharts,
-  wrapQtAppsHook,
+  qt6,
   cmake,
   python3,
   unstableGitUpdater,
@@ -29,13 +26,13 @@ stdenv.mkDerivation rec {
     cmake
     pkg-config
     python3
-    wrapQtAppsHook
+    qt6.wrapQtAppsHook
   ];
 
   buildInputs = [
-    qtbase
-    qtsvg
-    qtcharts
+    qt6.qtbase
+    qt6.qtsvg
+    qt6.qtcharts
   ];
 
   installPhase =

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -1404,8 +1404,6 @@ with pkgs;
 
   py65 = with python3.pkgs; toPythonApplication py65;
 
-  ripes = qt6Packages.callPackage ../applications/emulators/ripes { };
-
   rmg-wayland = callPackage ../by-name/rm/rmg/package.nix {
     withWayland = true;
   };

